tests/expected/function-contract/interior-mutability/cell_stub.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// kani-flags: -Zfunction-contracts
+
+// The objective of this test is to show that the contracts for double can be replaced as a stub within the contracts for quadruple.
+// This shows that we can generate kani::any() for Cell safely by breaking encapsulation
+
+// ---------------------------------------------------
+//        Abstraction Breaking Functionality
+// ---------------------------------------------------
+
+use std::cell::Cell;
+use std::mem::transmute;
+
+/// This exposes the underlying representation so that it can be added into a modifies clause within kani
+trait Exposeable<T: ?Sized> {
+    unsafe fn expose(&self) -> &T;
+}
+
+// This unsafe manipulation is valid due to Cell having the same underlying data layout as its internal T as explained here: https://doc.rust-lang.org/stable/std/cell/struct.Cell.html#memory-layout
+impl<T: ?Sized> Exposeable<T> for Cell<T> {
```

Comment:
Can you use `Cell::get` or `Cell::as_ptr` instead?

Reply:
This can be done for `Cell` and `UnsafeCell` and I've added two more test cases for this. I don't immediately see a way to do this for `OnceCell`. It won't extend to `RefCell` due to runtime borrow checks.

Reply:
You can use the address of the `OnceCell` itself.

Reply:
Using the address of `OnceCell` itself works! `RefCell::as_ptr()` would let you get the reference to the private field of `self.value` but it won't let you get the other fields `self.borrow` and `self.borrowed_at` that also are modified.

Reply:
In that case, you should also use the address of the `RefCell` object.

Reply:
Using the address of the `OneCell` or the `RefCell` will work for instrumenting the `check` generated function, but will not work for the `replace` generated function because they do not implement the `Arbitrary` trait, nor can they without breaking encapsulation. The current tests that exist can be replaced by a contract stub via `stub_verified`.

Reply:
Yes. Kani cannot stub contracts that do modifies variables of types that do not implement Arbitrary type. This is an orthogonal problem to interior mutability and to encapsulation.

It's OK if you want to add a test that shows how to handle all three problems at the same time; however, I would ask you to update the test documentation to make this clear.